### ISSUE TITLE
Problem with the name of ship and troops

### ISSUE THREAD

**Issue by naashil** [Wolferos/Hearts-of-Iron-IV-The-Great-War#199]
Hello, i love your mod but i juste want to say that i've big problemes with the name of the unit.

1) All my german ships have no name (really nothing, that's disturbing...)

![image](https://user-images.githubusercontent.com/53495290/62165417-afb86980-b31e-11e9-942e-f6f6e053440e.png)

2) when a new ship is coming out, he has the name of "modern" battleship of the second world war (for exemple : Tirpitz) That's not wery history friendly.....

![image](https://user-images.githubusercontent.com/53495290/62165817-919f3900-b31f-11e9-99a8-dbf376e89106.png)

3) All my new troops coming out have a generik name like "12 Infanterie Division" and i'm not abble to change it. 

![image](https://user-images.githubusercontent.com/53495290/62165490-e1313500-b31e-11e9-91ea-e9d82c667001.png)

When i load a game, all th ships have the same name... 

![image](https://user-images.githubusercontent.com/53495290/62165623-28b7c100-b31f-11e9-9d29-baec88dca2a4.png)

So if you can just fix this name issues to improve historical immersion, it would be great. 

Sorry my english is bad, i'm an Alsacian!

**Comment by sgerhardt** [comment]
Looks like the ship names are in the txt files, e.g.
https://github.com/Wolferos/Hearts-of-Iron-IV-The-Great-War/blob/develop/mod/thegreatwar/history/units/GER_1910.txt#L1036-L1038 

So it must be a problem with correctly loading them.

I wonder if this broke with the Man The Guns DLC? It looks like some tags were added as part of that, e.g. "fleet", "naval_base", "task_force", after comparing with the base game's naval files.

I wonder if the fix would just be a matter of restructuring these files to include those tags...

**Comment by naashil** [comment]
Thank you for your response, 

I don't know, do you think i have to reinstall the mod?? 

I've just desactivated the "Man the Gun" DLC and it does not work too... 

I really want to play this mod but with this name (ship and field army) issues it's not fun at all.. 

Hope you can help me, I don't know what to do

**Comment by sgerhardt** [comment]
@naashil This will require a code change to the great war mod. See https://github.com/Wolferos/Hearts-of-Iron-IV-The-Great-War/pull/203 which is a proof of concept of what needs to be done. I just updated it for the German Hochseeflotte. It would require updates for ALL the ships in the game... which will be a fairly large undertaking. I wanted to make sure one of the repo owners was OK with my approach first before anyone spends a bunch of time updating those files.

**Comment by naashil** [comment]
Yes, ok I understand, I've just looked the proof of concept. Hope that one of the repo owner will answer soon. Thank you for fixing this issues and I wich you good luck with the huge work that have to be done...
Big up ! 😉

**Comment by sgerhardt** [comment]
@naashil https://github.com/Wolferos/Hearts-of-Iron-IV-The-Great-War/pull/204 should fix the rest of them.

**Comment by naashil** [comment]
Ok nice, thank you for this work. And how can I test it in game?

**Comment by sgerhardt** [comment]
You can test by :
1. “git checkout” of my branch to get the files on your machine.

2. From the hearts of iron launcher, select Mod Tools and create the mod using the directory from step one.

3. Run the mod and test out any nation with a navy 😄🚢

Let me know what you think

**Comment by naashil** [comment]
@sgerhardt Hi!

So, firstly i want to thank you for your work and your explanation for installing the mod. The name of the fleet is good and the name of each already existing boat is good too. 

But, when a new boat is produced, i've still the "modern" name of the boat comming out (like Tirpitz etc...) Can we do something to fix this part? 
As you can see on the picture bellow, the new boats still have "modern" name.

![image](https://user-images.githubusercontent.com/53495290/68693205-26e4d700-0577-11ea-8b63-c94b0a2f943a.png)

Moreover, if i want to creat un new boat design, i'm not abble to give him generic name (like imperial germany names etc...) (see picture bellow) 

![Sans titre](https://user-images.githubusercontent.com/53495290/68696246-e1c3a380-057c-11ea-806d-62a45d37a6b0.png)

Warning : this consern not only the boat but all the field army. New units have generic names and are note history friendly. I believe that the field army should have better name when 1WW mod cam out. That's a huge work to do.. I'm sorry I really want to help you but I'm very very bad in coding...  

Once more, sorry for my particulary bad english x) 

Thx for your work, hop the following can be done.

**Comment by sgerhardt** [comment]
@naashil We can look at fixing the other issues too. 

To control scope, does https://github.com/Wolferos/Hearts-of-Iron-IV-The-Great-War/pull/204 fix the issue of initial ship names and fleets not showing and not introduce any new problems? 

If so, we can at least fix the initial names of ships and fleets now, then we can start working on the other problems you noticed.

**Comment by naashil** [comment]
I've seen no problem with Germany. However, with France, Austria-Hungary etc... there's still this problem with the name of ship.

Good luck for the following

**Comment by sgerhardt** [comment]
@naashil Are you sure you're using the code from https://github.com/Wolferos/Hearts-of-Iron-IV-The-Great-War/pull/204 ? I'm able to load ships for France, Austria-Hungary, Russia, Japan, etc. as I'd expect.

**Comment by naashil** [comment]
I tested with the file in :
sgerhardt/HeartsOfIronModTools and than your branch. I clicked on dowload (green button) and installed what's in this file. 

It's maybe me. I did something wrong?

**Comment by sgerhardt** [comment]
@naashil You'll need [this ](https://github.com/sgerhardt/Hearts-of-Iron-IV-The-Great-War/tree/fix_ships) branch locally.

You shouldn't need the mod tools files, that was only used to generate the files in the above branch.

**Comment by naashil** [comment]
With the new UI of paradox I'm not abble to install this. I'm noobish sorry but I can not do it. 

I reinstalled the mod through steam and the name of the fleet is working for the german hochseeflotte in Wilhelmshaven but in the other harbour it doesn't work. 
For other countries it does not work... 

Sorry but I can not help you more :/

**Comment by sgerhardt** [comment]
No worries, @naashil - it merged today and was included in the latest release. You should have access to it via steam as soon as it’s uploaded.

I’ll take a look at the other problems you mentioned this weekend.

**Comment by naashil** [comment]
@sgerhardt thank you and good luck !  I'm in a hurry to play WW 1 mod  😀

**Comment by sgerhardt** [comment]
I'm no naval historian, so there's a few questions that would need to be answered.

1. What names should we use for the various ship classes?

2. What fallback names should be used after all the unique names are used? For example, these are usually generic names, like T-"Insert incrementing number here" for torpedo boats.

I guess here's a good starting point... https://en.wikipedia.org/wiki/German_ocean-going_torpedo_boats_and_destroyers_of_World_War_I

**Comment by naashil** [comment]
Hello, I just tested the mod since you improved it,

So, There are some improvement, for the navy almost all the existing boats have the good name. The boats in production have the good name too. But it remains one thing, when I build a new boat it still has the wrong name :/

Thank you for this great mod

**Comment by sgerhardt** [comment]
@naashil The proof of concept fix for ship names is not fully released yet. You’d have to test with the unreleased code. Also, you’d need to test with one of the obsolete ships in https://github.com/Wolferos/Hearts-of-Iron-IV-The-Great-War/pull/206 to see the new correct names.

**Comment by naashil** [comment]
Yup, I'm only saying that's a good start 😉 good luck for the following
